From 75a167dac177b4c08b49955b74166d42b287c019 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Tue, 31 May 2011 00:26:57 +0800
Subject: [PATCH 984/998] ENGR00144219 Fix vpu caused system hang due to clock gated off on mx53

The specific stream 'mpeg2_720x576.mpg' in test lib will make system
hang when deblock/dering is enabled with following cmd:
./mxc_vpu_test.out -D "-i mpeg2_720x576.mpg -f 4 -e 1 -d 1"

This is because vpu decoding is dependent on emi_intr_clk[0], but
this dependency isn't added.
So adding this in mx53_clocks_init() function.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 arch/arm/mach-mx5/clock.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-mx5/clock.c b/arch/arm/mach-mx5/clock.c
index 46c8149..c051fd8 100644
--- a/arch/arm/mach-mx5/clock.c
+++ b/arch/arm/mach-mx5/clock.c
@@ -4861,6 +4861,8 @@ int __init mx53_clocks_init(unsigned long ckil, unsigned long osc, unsigned long
 	esdhc3_clk[0].get_rate = _clk_esdhc3_get_rate;
 	esdhc3_clk[0].set_rate = _clk_sdhc3_set_rate;
 
+	vpu_clk[2].secondary = &emi_intr_clk[0];
+
 	ata_clk[1].secondary = &tmax3_clk;
 
 #if defined(CONFIG_USB_STATIC_IRAM) \
-- 
1.7.4.1

